earthfile2llb/converter.go

```diff
@@ -2416,6 +2428,41 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 }
 
+// detectAWSCredentials determines whether AWS envs are set and/or ~/.aws
+// exists. These are then used when fetching the secrets during a build.
+func (c *Converter) detectAWSCredentials(ctx context.Context) ([]llb.RunOption, []string, error) {
+
+	credsAvail, err := awsutil.CredsAvailable()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to check for AWS credentials")
+	}
+
+	if !credsAvail {
+		return nil, nil, errors.New("AWS credentials not found in environment or ~/.aws")
+	}
+
+	var (
+		runOpts   = []llb.RunOption{}
+		extraEnvs = []string{}
+	)
+
+	// Add LLB secrets for each of the typical secrets which will then be
+	// sourced from the environment during lookup.
+	for _, secretName := range secretprovider.AWSCredentials {
+		secretPath := path.Join("/run/secrets", secretName)
+		secretOpts := []llb.SecretOption{
+			llb.SecretID(c.secretID(secretName)),
+			llb.SecretFileOpt(0, 0, 0444),
+		}
+		runOpts = append(runOpts, llb.AddSecret(secretPath, secretOpts...))
+		if envName, ok := secretprovider.AWSEnvName(secretName); ok {
+			extraEnvs = append(extraEnvs, fmt.Sprintf("%s=\"$(cat %s)\"", envName, secretPath))
```

Comment:
Is there a good way to make this a bit more uniform? Like if one developer uses `~/.aws`, but another might use the env vars, could we normalize into one of the two representations every time, so that it always shows up the same way within Earthly? It'll create better consistency.

Reply:
Can you elaborate on what you mean by "uniform"?

The same special secret names (below) are added here as env vars whether they're present in the host's env or config file. They are already normalized, in a sense. 

https://github.com/earthly/earthly/blob/mh%2Faws-creds/util/llbutil/secretprovider/aws_creds.go#L17

Reply:
Ah ok you're already doing this - awesome! I missed the normalization part.